### PR TITLE
feat: add project role management to SCIM user updates

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #XXXX

### Description:
Adds support for managing user project roles through the SCIM API. This implementation enables:

1. Parsing and validation of role IDs in the format `projectUuid:roleUuid`
2. Updating both system roles and custom roles for projects when a user is updated via SCIM
3. Proper validation of role arrays to ensure:
   - Exactly one organization role is provided
   - No duplicate project roles exist
   - All role values are valid

The changes include comprehensive tests for the new functionality, covering role parsing, validation, and the update process for both organization and project-level roles.

#### Update - add project editor role

Screenshot 2025-11-05 at 15 03 26

#### Update without `roles` property. Success and keeps existing roles
Screenshot 2025-11-05 at 15 04 14

#### Errors
Screenshot 2025-11-05 at 15 03 59
Screenshot 2025-11-05 at 15 03 48
Screenshot 2025-11-05 at 15 02 56

### CODE REVIEW

**Comment by ZeRego** [comment]
* **#17925** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17925?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17914** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17914?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17899** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17899?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17895** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17895?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17894** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17894?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17894?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#17879** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17879?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17872** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17872?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17851** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/17851?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/backend/src/ee/services/ScimService/ScimService.ts [hunk 1]
```diff
@@ -1432,6 +1476,84 @@ export class ScimService extends BaseService {
...
+        projectUuid?: string;
+    } {
+        if (!roleId.includes(':')) {
+            return {
+                roleUuid: roleId,
+                projectUuid: undefined,
+            };
+        }
+
+        const colonIndex = roleId.indexOf(':');
```
Comment (rephus) [inline packages/backend/src/ee/services/ScimService/ScimService.ts:1490]: Why not doing split?

Comment (ZeRego) [reply packages/backend/src/ee/services/ScimService/ScimService.ts:1490]: Handles edge case scenario of multiple `:` in string. Although it should not happen, there is a unit test to cover it.
